chat/src/main/java/io/skygear/plugins/chat/ChatContainer.java

```diff
@@ -595,13 +610,41 @@ public void onQuerySuccess(Record[] records) {
                     return;
                 }
 
-                List<UserConversation> userConversations = null;
+                final List<UserConversation> userConversations = new LinkedList<>();;
                 if (records != null && records.length > 0) {
-                    userConversations = new LinkedList<>();
                     for (Record eachRecord : records) {
                         userConversations.add(new UserConversation(eachRecord));
                     }
                 }
+                if (getLastMessages) {
+                    List<String> messageIds = new ArrayList<String>();
+                    for (UserConversation uc : userConversations) {
+                        String mid = uc.getLastReadMessageId();
+                        if (mid != null) {
+                            messageIds.add(mid);
+                        }
+                    }
+                    getMessagesByIds(messageIds, new GetCallback<List<Message>>() {
+                        @Override
+                        public void onSucc(@Nullable List<Message> messages) {
+                            Map<String, Message> mMap = new HashMap<String, Message>();
+                            for (Message m : messages) {
+                                mMap.put(m.getId(), m);
+                            }
+                            for (UserConversation uc: userConversations) {
+                               if (uc.getLastReadMessageId() != null) {
+                                   uc.lastMessage = mMap.get(uc.getLastReadMessageId());
+                               }
+                            }
+                            callback.onSucc(userConversations);
+                        }
+
+                        @Override
+                        public void onFail(@Nullable String failReason) {
+                            callback.onFail(failReason);
+                        }
+                    });
+                }
                 callback.onSucc(userConversations);
```

Comment:
@rickmak The `callback.onSucc()` may be called twice when `getLastMessages` is set to `true`. That should not be the expected behavior, right?